dirty_equals/_other.py

```diff
@@ -542,3 +543,166 @@ class Foo:
 
     def _post_init(self) -> None:
         self.strict = True
+
+
+class IsEnum(DirtyEquals[Enum]):
+    """
+    Checks if an instance is an Enum.
+
+    Inherits from [`DirtyEquals`][dirty_equals.DirtyEquals].
+
+    ```py title="IsEnum"
+    from enum import Enum, auto
+    from dirty_equals import IsEnum
+
+    class ExampleEnum(Enum):
+        a = auto()
+        b = auto()
+
+    a = ExampleEnum.a
+    assert a == IsEnum
+    assert a == IsEnum(ExampleEnum)
+    ```
+    """
+
+    def __init__(self, enum_cls: type[Enum] = Enum):
+        """
+        Args:
+            enum_cls: Enum class to check against.
+        """
+        self._enum_cls = enum_cls
+
+    def equals(self, other: Any) -> bool:
+        return isinstance(other, self._enum_cls)
+
+
+class IsEnumType(DirtyEquals[Enum]):
```

Comment:
On review, I'm not sure that all these types are really necessary - `IsEnum` seems useful, but this seems like overkill.

LMK if you disagree.

Reply:
I didn't (and still don't) have a direct use case to justify them. Mostly tried to replicate the behaviour of `IsDataclass` and `IsDataclassType` (as in #68).

Let me trim them from this PR